### ISSUE TITLE
bad JSON files

### ISSUE THREAD

**Issue by sc00n** [cph-cachet/carp.sensing-flutter#305]
As dataEndPoint we use FileDataEndPoint. Unfortenatelly, the JSONs in this file are broken. We often (almost always) see the following pattern:

```
{"carp_header":{"study_id":"StudyMerijn","device_role_name":"masterphone","trigger_id":"13","user_id":"43190a10-9019-11ed-a6a5-051decb2df1c","start_time":"2023-01-09T12:31:31.452237Z","data_format":{"namespace":"dk.cachet.carp","name":"memory"}},"carp_body":{"id":"8bed24b0-9019-11ed-8065-ffe756b129c4","timestamp":"2023-01-09T12:31:31.451719Z","free_physical_memory":204419072,"free_virtual_memory":1940180992}}{"carp_header":{"study_id":"StudyMerijn","device_role_name":"masterphone","trigger_id":"4","user_id":"43190a10-9019-11ed-a6a5-051decb2df1c","start_time":"2023-01-09T12:31:31.453709Z","data_format":{"namespace":"dk.cachet.carp","name":"pedometer"}},"carp_body":{"id":"8bed72d0-9019-11ed-a864-97ae77e357e3","timestamp":"2023-01-09T12:31:31.453545Z","step_count":13389}}{"carp_header":{"study_id":"StudyMerijn","device_role_name":"masterphone","trigger_id":"12","user_id":"43190a10-9019-11ed-a6a5-051decb2df1c","start_time":"2023-01-09T12:31:31.454797Z","data_format":{"namespace":"dk.cachet.carp","name":"connectivity"}},"carp_body":{"id":"8bed99e0-9019-11ed-a9bb-a3362f2bf6ee","timestamp":"2023-01-09T12:31:31.454645Z","connectivity_status":"mobile"}}
,
 
,
 
,
```

Three entries, and only then we see the three commas. The commas obviously should be between the entries. This makes it very difficult to read the files. Sometimes its more or less entries that are posted without commas.

**Comment by sc00n** [comment]
created a pull request to fix this: https://github.com/sc00n/carp.sensing-flutter/pull/1

**Comment by bardram** [comment]
I will try to look into this. 

But, since v. 0.40 CAMS have supported the [`SQLiteDataEndPoint`](https://pub.dev/documentation/carp_mobile_sensing/latest/domain/SQLiteDataEndPoint-class.html) data endpoint which stores data in a local SQLite database using the [`SQLiteDataManager`](https://pub.dev/documentation/carp_mobile_sensing/latest/managers/SQLiteDataManager-class.html).

This method is much more stable (and fast) than the file endpoint and I recommend using this instead. The data format is the same, just stored in a table in SQLite.

**Comment by bardram** [comment]
Fixed and released as carp_mobile_sensing 0.40.11
